packages/react/src/menu/popup/MenuPopup.tsx

```diff
@@ -49,6 +50,8 @@ const MenuPopup = React.forwardRef(function MenuPopup(
     onOpenChangeComplete,
   } = useMenuRootContext();
   const { side, align, floatingContext } = useMenuPositionerContext();
+  const contextMenuContext = useContextMenuRootContext();
+  const hasContextMenuParent = Boolean(contextMenuContext) && !nested;
```

Comment:
In the Menubar PR I implemented a more universal solution to determine the menu's parent and its capabilities: https://github.com/mui/base-ui/pull/1684/files#diff-6da7b84ff007a487d62a9d48ba9fb32ad4b286ef666fd3d0419b14e9ebbe472a.

Reply:
Context menus don't have any parents though, so I think this should suffice?

Reply:
I'd like to avoid having all kinds of boolean values like `nested`, `isInContextMenu`, `isInMenubar`, (possibly) `isInNavigationMenu`. The approach I took in the Menubar PR lets a menu part know what kind of menu it's in in a standard way and provides a type-safe way to access the parent context (without having to check for its existence)

Reply:
Alright 👍 I'll wait for Menubar to be merged and then use its logic for `Menu`